### PR TITLE
Avoid errors caused by `more-dropdown` on issues

### LINKED ISSUE #3641: `GetCurrentBranch` Stopped working on issues
This is a major problem

Test on https://github.com/sindresorhus/refined-github/issues/3638 or even this issue.

### PR DESCRIPTION
**Pull request by yakov116**
1. LINKED ISSUES:
   Fixes #3641 

2. TEST URLS:
   #3641 

3. SCREENSHOT:
   None

### CODE REVIEW

File: source/github-helpers/index.ts [hunk 1]
```diff
@@ -21,9 +21,15 @@ Tested on isRepoTree, isBlame, isSingleFile, isEditFile, isSingleCommit, isCommi
...
 Example tag content on private repositories https://github.com/private/private/commits/master.atom?token=AEAXKWNRHXA2XJ2ZWCMGUUN44LM62
 */
-export const getCurrentBranch = (): string => {
+export const getCurrentBranch = (): string | undefined => {
 	// .last needed for #2799
-	return new URL(select.last<HTMLLinkElement>('[type="application/atom+xml"]')!.href)
+	const atom = select.last<HTMLLinkElement>('[type="application/atom+xml"]');
+	// The atom URL is not available on `isIssue` #3641
+	if (!atom) {
+		return undefined;
```
Comment (fregante) [inline source/github-helpers/index.ts:29]:
```suggestion
		return;
```

Doesn’t work? Or just `void` as the type?

Comment (fregante) [reply source/github-helpers/index.ts:29]: I’m on mobile now but it sounds strange that there’s no atom feed on issues at all

Comment (yakov116) [reply source/github-helpers/index.ts:29]: I agree! This was working all along.

File: source/github-helpers/index.ts [hunk 2]
```diff
@@ -21,9 +21,15 @@ Tested on isRepoTree, isBlame, isSingleFile, isEditFile, isSingleCommit, isCommi
 Example tag content on public repositories: https://github.com/sindresorhus/refined-github/commits/branch-or-commit/even/with/slashes.atom
 Example tag content on private repositories https://github.com/private/private/commits/master.atom?token=AEAXKWNRHXA2XJ2ZWCMGUUN44LM62
 */
-export const getCurrentBranch = (): string => {
+export const getCurrentBranch = (): string | undefined => {
 	// .last needed for #2799
-	return new URL(select.last<HTMLLinkElement>('[type="application/atom+xml"]')!.href)
+	const atom = select.last<HTMLLinkElement>('[type="application/atom+xml"]');
```
Comment (notlmn) [inline source/github-helpers/index.ts:26]: Could be more descriptive as

```suggestion
	const feedLink = select.last<HTMLLinkElement>('[type="application/atom+xml"]');
```

Or `repoFeedLink` to be a bit more verbose?

File: source/github-helpers/index.ts [hunk 3]
```diff
@@ -21,9 +21,15 @@ Tested on isRepoTree, isBlame, isSingleFile, isEditFile, isSingleCommit, isCommi
...
+export const getCurrentBranch = (): string | undefined => {
 	// .last needed for #2799
-	return new URL(select.last<HTMLLinkElement>('[type="application/atom+xml"]')!.href)
+	const feedLink = select.last<HTMLLinkElement>('[type="application/atom+xml"]');
+	// The feedLink is not available on `isIssue` #3641
+	if (!feedLink) {
+		return;
+	}
+
+	return new URL(atom.href)
```
Comment (fregante) [inline source/github-helpers/index.ts:32]:
```suggestion
	return new URL(feedLink.href)
```

**Review by fregante — APPROVED** [review]

**Comment by yakov116** [comment]
@fregante when you approve can I assume the title is good?

File: source/github-helpers/github-url.ts [hunk 4]
```diff
@@ -43,7 +43,7 @@ export default class GitHubURL {
 
 		const filePath = ambiguousReference.slice(1).join('/');
 
-		const currentBranch = getCurrentBranch();
+		const currentBranch = getCurrentBranch()!;
 		const currentBranchSections = currentBranch.split('/');
 		if (
 			ambiguousReference.length === 1 || // Ref has no slashes
```
Comment (fregante) [inline source/github-helpers/github-url.ts:49]:
```suggestion
		const currentBranch = getCurrentBranch();
		const currentBranchSections = currentBranch?.split('/');
		if (
			!currentBranch || // Current page does not have a branch; GitHubURL shouldn’t really be used here
			ambiguousReference.length === 1 || // Ref has no slashes
```

File: source/github-helpers/github-url.ts [hunk 5]
```diff
@@ -44,8 +44,9 @@ export default class GitHubURL {
 		const filePath = ambiguousReference.slice(1).join('/');
 
 		const currentBranch = getCurrentBranch();
-		const currentBranchSections = currentBranch.split('/');
+		const currentBranchSections = currentBranch?.split('/');
 		if (
+			!currentBranch || // Current page does not have a branch; GitHubURL probably shouldn’t be used here
```
Comment (fregante) [inline source/github-helpers/github-url.ts:49]: This should fix the types
```suggestion
			!currentBranchSections || // Current page does not have a branch; GitHubURL probably shouldn’t be used here
```

**Comment by yakov116** [comment]
https://github.com/sindresorhus/refined-github/blob/10dc4ac56c299c011733483a20725f3618a05175/source/github-helpers/github-url.ts#L65

Here is the issue

**Comment by fregante** [comment]
Just fix the type issue with `!` or something.

I checked all the features where we use GitHubURL and all of them have a branch. In `forked-to` we specifically avoid errors by limiting the usage of GitHubURL to pages with a branch, even if the feature is enabled on `isRepo` https://github.com/sindresorhus/refined-github/blob/d7d700a20befb552f9f64e15504b845dbe04e699/source/features/forked-to.tsx#L34-L36

In another feature we also specifically mention the incompatibility so we don't use GitHubURL:
 https://github.com/sindresorhus/refined-github/blob/b7d31c6ae88ce48f4f5fa2716716a40e32eeea1b/source/features/comments-time-machine-links.tsx#L77

**Comment by fregante** [comment]
Actually the ideal solution would be to throw an error when the page doesn't have a current branch to avoid silent errors.

File: source/github-helpers/github-url.ts [hunk 6]
```diff
@@ -44,8 +44,9 @@ export default class GitHubURL {
 		const filePath = ambiguousReference.slice(1).join('/');
 
 		const currentBranch = getCurrentBranch();
```
Comment (fregante) [inline source/github-helpers/github-url.ts:46]: How about this?
```suggestion
		const currentBranch = getCurrentBranch();
		if (!currentBranch) {
			throw new Error('GitHubURL can only be used on pages with a branch/reference.')
		}
```

**Comment by fregante** [comment]
> @fregante when you approve can I assume the title is good?

Not always. Generally try to make the commit titles readable by users. "returns undefined" doesn't explain how this helps the user. In this case the specific user-facing bug being solved is ``Avoid errors caused by `more-dropdown` on issues`` or something along those lines

**Review by fregante — APPROVED** [review]

**Pull request merged**